projects/fw_103/src/main.c

```diff
@@ -0,0 +1,68 @@
+#include "adc.h"
+#include "delay.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+
+// A6 input, B2 input, log output
+
+// callback func, takes in the channel and reads ADC data from channel into local var
+// then logs the var data
+static void prv_btn_interrupt_handler(const GpioAddress *address, void *context) {
+  AdcChannel *in_channel = (AdcChannel *)context;
+  uint16_t in_data = 0;
+
+  adc_read_raw(*in_channel, &in_data);
+  LOG_DEBUG("Input ADC Data: %d\n", in_data);
+}
+
+int main(void) {
+  // initialize libraries
+  interrupt_init();
+  soft_timer_init();
+  gpio_init();
+  gpio_it_init();
+
+  // assign ports and pins for ADC input and Button input
+  GpioAddress adc_button_addresses[] = {
+    [ADC_IN] = { .port = GPIO_PORT_A, .pin = 6 },  //
+    [BTN_IN] = { .port = GPIO_PORT_B, .pin = 2 },  //
```

Comment:
I like the spirit, but these constants `ADC_IN` and `BTN_IN` aren't defined, so this won't compile. Perhaps two variables would be better instead?